src/main/kotlin/ar/edu/unahur/obj2/caralibro/Usuario.kt

```diff
@@ -1,11 +1,60 @@
 package ar.edu.unahur.obj2.caralibro
 
 class Usuario {
-  val publicaciones = mutableListOf<Publicacion>()
+    val publicaciones = mutableListOf<Publicacion>()
+    val amigos = mutableListOf<Usuario>()
+    val listaPermitidos = mutableListOf<Usuario>()
+    val listaExcluidos = mutableListOf<Usuario>()
 
-  fun agregarPublicacion(publicacion: Publicacion) {
-    publicaciones.add(publicacion)
-  }
+    fun agregarPublicacion(publicacion: Publicacion) {
+        publicaciones.add(publicacion)
+    }
+
+    fun agregarAListaPermitidos(usuario: Usuario) = this.listaPermitidos.add(usuario)
+
+    fun agregarAListaExcluidos(usuario: Usuario) = this.listaExcluidos.add(usuario)
+
+    fun espacioDePublicaciones() = publicaciones.sumBy { it.espacioQueOcupa() }
+
+    fun darLike(publicacion: Publicacion) {
+        if (this.dioLikeEn(publicacion)) {
+            throw Exception("Ya le has dado like a esta publicacion")
+        }
+        publicacion.likes += 1
+        publicacion.usuariosQueDieronLike.add(this)
+    }
+
+    fun dioLikeEn(publicacion: Publicacion) = publicacion.usuariosQueDieronLike.contains(this)
+
+    fun agregarAmigo(usuario: Usuario) {
+        this.amigos.add(usuario)
+    }
+
+    fun esMasAmistosoQue(usuario: Usuario) = this.amigos.count() > usuario.amigos.count()
+
+    fun puedeVerPublicacion(publicacion: Publicacion, usuarioQueDeseaVerPublicacion: Usuario) =
+        publicacion.puedeSerVistaPor(this, usuarioQueDeseaVerPublicacion)
+
+    fun cantidadLikes() = this.publicaciones.map { it.likes }.sum()
+
+    fun amigoMasPopular() = this.amigos.maxByOrNull { it.cantidadLikes() }
+
+    fun esMejorAmigo(usuarioMejorAmigo: Usuario): Boolean {
+        return this.amigos.find { usuario: Usuario -> usuario == usuarioMejorAmigo } == usuarioMejorAmigo
+    }
```

Comment:
Esto no es lo que se pedía.
> 5. Determinar los mejores amigos de un usuario: el conjunto de sus amigos que pueden ver todas sus publicaciones.

Una solución podría ser un método  auxiliar que te diga si un usuario puede ver todas las publicaciones del otro.
Y después filtran todos los amigos que cumplan esta condición.
```suggestion
    fun puedeVerTodasMisPublicacion(usuario: Usuario) = logica que calcule eso....
    fun mejoresAmigos() = this.amigos.filter {this.puedeVerTodasMisPublicacion(it)}
```